src/main/java/codeu/model/data/ConversationActivity.java

```diff
@@ -0,0 +1,46 @@
+// ConversationActivity
+// Represents a Message sent in a conversation. Keeps track of the message sent,
+// the conversation it was sent in, and the user who sent the message and the
+// time the message was sent for displaying the activity in the activity feed.
+
+package codeu.model.data;
+
+import codeu.model.data.Conversation;
+import codeu.model.data.Message;
+import java.util.UUID;
+
+public class ConversationActivity extends Activity {
+ 
+  private String conversationTitle;
+  private String messageContent;
+
+  private final int messageMaxLen = 30;   //max length for message preview
+
+  // initialize activity based on message and conversation title
+  public ConversationActivity(Message newMessage, String conversationTitle) {
```

Comment:
That makes sense, and I think I'll go with 2. since it's closer to what I have now and there shouldn't need to be a lot of activities needing to be stored at once whereas there may be a lot of messages. And I will change it to just save the message preview; I think I was thinking before that the message preview length could change but I don't think that concern is important.